share/completions/yadm.fish

```diff
@@ -0,0 +1,131 @@
+# Completions for yadm v2
+
+# Print an optspec for argparse to handle yadm's options
+# that are independent of any subcommand. Borrowed from git completion file.
+function __fish_yadm_global_optspecs
+    string join \n v-version h/help C= c=+ 'e-exec-path=?' H-html-path M-man-path I-info-path p/paginate \
+        P/no-pager o-no-replace-objects b-bare G-git-dir= W-work-tree= N-namespace= S-super-prefix= \
+        l-literal-pathspecs g-glob-pathspecs O-noglob-pathspecs i-icase-pathspecs \
+        Y/yadm-dir= R-yadm-repo= n-yadm-config= E-yadm-encrypt= a-yadm-archive= B-yadm-bootstrap=
+end
+
+# Config values accepted by `yadm config`
+# See `yadm gitconfig` for setting git config values
+function __fish_yadm_config_keys
+    yadm introspect configs | string replace \n " "
```

Comment:
I don't believe this `string replace` does what you think it does.

I would imagine, and do correct me if I'm wrong, that `yadm introspect configs` prints one config per line, so they look like

```
configone
configtwo
```

What you assume happens when you have a completion like

```fish
complete -c yadm -a '(__fish_yadm_config_keys)'
```

is that the inner command's output is placed directly into the `-a` argument, so this would have to look like

```
complete -c yadm -a 'configone configtwo'
```

 That's not what happens.

Like every other fish command substitution (that doesn't use `string split`), it is split, on newlines. The command's output is taken, each line is used as a separate argument.

So, if the configs are printed one per line, there is no need to change anything, and you could directly run

```fish
complete -c yadm -a '(yadm introspect configs)'
```

Or put it into this function *without the replace*.

If that for some reason doesn't work, I would love to see some example output of `yadm introspect configs`, and some example usage for `yadm config`

Reply:
You're absolutely right, `yadm introspect configs` prints one config per line and the `string replace` is redundant. Thanks for the review !